xla/tests/convert_test.cc

```diff
@@ -914,6 +913,107 @@ XLA_TYPED_TEST(ConvertTestF16, ConvertF8e5m2F16RoundtripExhaustive4) {
   this->ComputeAndCompare(&builder, {}, ErrorSpec(0.));
 }
 
+// ----- F8E4M3
+
+XLA_TEST_F(ConvertTest, ConvertF16F8e4m3Roundtrip) {
+  // Convert from FP16 to FP8, then back to FP16
+  XlaBuilder builder(TestName());
+  float nan = std::numeric_limits<float>::quiet_NaN();
+  float inf = std::numeric_limits<float>::infinity();
+
+  struct TestCase {
+    float input;
+    float expected_roundtrip;
+  } test_cases[] = {
+      // clang-format off
+      {0.0, 0.0},
+      {-0.0, -0.0},
+      {1.0, 1.0},
+      {-1.0, -1.0},
+      {nan, nan},
+      {-nan, -nan},
+      {inf, inf},
+      {-inf, -inf},
+      // clang-format on
+      {0x1.1p0, 0x1p0},      // Round-to-even down
+      {0x1.3p0, 0x1.4p0},    // Round-to-even up
+      {0x1.Ep7, 0x1.Ep7},    // Max value
+      {0x1.EFCp7, 0x1.Ep7},  // Largest number that doesn't overflow
+      {0x1.Fp7, inf},        // Smallest number that overflows
+      {0x1p8, inf},          // Overflow
+      {0x1p-6, 0x1p-6},      // Smallest normal
+      {0x0.2p-6, 0x0.2p-6},  // Smallest denormal
+      {0x0.Ep-6, 0x0.Ep-6},  // Largest denormal
```

Comment:
Update test to match recent changes in xla/tests/convert_test.cc https://github.com/openxla/xla/pull/17437

Reply:
Updated the tests (ConvertF16 + Roundtrip tests).

One additional test was added recently `DISABLED_ON_CPU(ConvertF32F8e5m2Roundtrip)`. Need to add similar test for e4m3 and e3m4

Reply:
Added F32 test:
- `DISABLED_ON_CPU(ConvertF32F8e4m3Roundtrip)`
- `DISABLED_ON_CPU(ConvertF32F8e3m4Roundtrip)`